File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlLogicalOperator.cs
namespace T1.SqlSharp.Expressions;

public class SqlLogicalOperator : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.LogicalOperator;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_LogicalOperator(this);
    }

    public LogicalOperator Value { get; set; }
    public string ToSql()
    {
        return Value.ToSql();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlLogicalOperator.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { LogicalOperator } from './LogicalOperator';
import { logicalOperatorToSql } from './LogicalOperatorExtensions';

export class SqlLogicalOperator implements ISqlExpression {
    SqlType: SqlType = SqlType.LogicalOperator;
    Span: TextSpan = new TextSpan();
    Value!: LogicalOperator;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_LogicalOperator(this);
    }

    ToSql(): string {
        return logicalOperatorToSql(this.Value);
    }
} 